v2/tools/generator/internal/reporting/type_catalog_report.go

```diff
@@ -269,41 +307,45 @@ func (tcr *TypeCatalogReport) asDefinitionToInline(
 	return nil, false
 }
 
-// asShortName returns a short name for the type, for use when it's inlined
-func (tcr *TypeCatalogReport) asShortName(def astmodel.TypeDefinition) string {
-	if _, r := astmodel.AsResourceType(def.Type()); r {
-		return "Resource"
-	}
-
-	if _, ok := astmodel.AsEnumType(def.Type()); ok {
-		return "enum"
-	}
-
-	if m, ok := astmodel.AsMapType(def.Type()); ok {
+func (tcr *TypeCatalogReport) asShortNameForType(t astmodel.Type, currentPackage astmodel.PackageReference) string {
+	// We switch on exact types because we don't want to accidentally unwrap a detail we need
+	switch t := t.(type) {
+	case *astmodel.OptionalType:
 		return fmt.Sprintf(
-			"map[%s]",
-			astmodel.DebugDescription(m.KeyType(), def.Name().PackageReference))
-	}
-
-	if a, ok := astmodel.AsArrayType(def.Type()); ok {
+			"*%s",
+			tcr.asShortNameForType(t.Element(), currentPackage))
+	case *astmodel.ArrayType:
 		return fmt.Sprintf(
 			"%s[]",
-			astmodel.DebugDescription(a.Element(), def.Name().PackageReference))
-	}
-
-	if one, ok := astmodel.AsOneOfType(def.Type()); ok {
+			tcr.asShortNameForType(t.Element(), currentPackage))
+	case *astmodel.MapType:
 		return fmt.Sprintf(
-			"oneOf[%d]",
-			one.Types().Len())
-	}
-
-	if all, ok := astmodel.AsAllOfType(def.Type()); ok {
+			"map[%s]%s",
+			tcr.asShortNameForType(t.KeyType(), currentPackage),
+			tcr.asShortNameForType(t.ValueType(), currentPackage))
+	case *astmodel.ResourceType:
+		return "Resource"
+	case *astmodel.EnumType:
+		return fmt.Sprintf(
+			"Enum (%s)",
+			tcr.asCount(len(t.Options()), "value", "values"))
+	case *astmodel.ObjectType:
+		return fmt.Sprintf(
+			"Object (%s)",
+			tcr.asCount(t.Properties().Len(), "property", "properties"))
+	case *astmodel.OneOfType:
 		return fmt.Sprintf(
-			"allOf[%d]",
-			all.Types().Len())
+			"OneOf (%s)",
+			tcr.asCount(t.Types().Len(), "choice", "choices"))
+	case *astmodel.AllOfType:
+		return fmt.Sprintf(
+			"AllOf (%s)",
+			tcr.asCount(t.Types().Len(), "choice", "choices"))
+	case astmodel.MetaType:
```

Comment:
Same question about base case here, although I suppose here if we fall through we just fall back to the debug description which seems reasonable.